openconfig_pyang/plugins/openconfig.py

```diff
@@ -721,6 +729,29 @@ def check_enumeration_style(ctx, stmt):
         err_add(ctx.errors, stmt.pos, "OC_ENUM_UNDERSCORES",
                 (enum.arg, enum.arg.upper()))
 
+  @staticmethod
+  def check_posix_pattern_equal(ctx, stmt):
+    """Check that OpenConfig's posix-pattern extension always equal in number to
+    pattern statements.
+
+    Args:
+      ctx: pyang.Context for validation
+      stmt: pyang.Statement representing a leaf or leaf-list
+          containing an enumeration
+    """
+    elemtype = stmt.search_one("type")
+    if elemtype is None:
+      return
+
+    patternCount = len(elemtype.search("pattern"))
```

Comment:
Does `elemtype.search` return `None` if there is no pattern statement? If so, doesn't `len(None)` throw a `TypeError`?

Reply:
Just checked that it always returns a list.